server/text_generation_server/models/custom_modeling/flash_llama_modeling.py

```diff
@@ -336,9 +338,21 @@ def __init__(self, index, prefix, config, weights):
             config=config,
             weights=weights,
         )
-        self.mlp = LlamaMLP(
-            prefix=f"{prefix}.mlp", config=config, weights=weights, index=index
-        )
+
+        # Only use FP8 when we are not using another quantizer. We should probably
+        # have nicer way of querying this, but for now it'll do.
+        weights_loader = weights.weights_loader
+        if (
+            can_use_fp8()
+            and isinstance(weights_loader, DefaultWeightsLoader)
+            and weights_loader.weight_class is UnquantizedWeight
```

Comment:
Doesn't this mean that we will always use fp8 for the MLP? I don't think we want that.
In my opinion it should be the other way around; wrap all the other linear layers with a context manager using UnquantizedWeight if we are in FP8 but that's a bit cumbersome.

I guess that's why the outputs changed.

Reply:
Changed around as you suggested (do nothing, unless `--quantize=fp8`, then exclude attention and piece embeddings).